packages/dds/sequence/src/intervalCollection.ts

```diff
@@ -1347,17 +1347,42 @@ export class IntervalCollection<TInterval extends ISerializableInterval>
             : serializedIntervals.intervals.map((i) => decompressInterval(i, serializedIntervals.label));
     }
 
+    private rebasePositionWithSegmentSlide(
+        pos: number,
+        seqNumberFrom: number,
+        localSeq: number
+    ): number | undefined {
+        if (!this.client) {
+            throw new LoggingError("mergeTree client must exist");
+        }
+        const { clientId } = this.client.getCollabWindow();
+        const { segment, offset } = this.client.getContainingSegment(pos, undefined, localSeq, seqNumberFrom, clientId);
+
+        // if segment is undefined, it slid off the string
+        assert(segment !== undefined, "No segment found");
```

Comment:
I'm a little surprised that the logic that used to happen in `rebasePosition` doesn't still need to happen in some form here:

```typescript
        if (segment === undefined && offset === undefined) {
            // getContainingSegment will only return non-removed segments. This means the position was past
            // all non-removed segments in the tree, so we take the last one instead as an approximation.
            let finalSegment: IMergeNode = this._mergeTree.root;
            while (!finalSegment.isLeaf()) {
                finalSegment = finalSegment.children[finalSegment.childCount - 1];
            }
            segment = finalSegment;
            offset = 0;
        }
```

If you've confirmed fuzz tests are passing, it's possible that the comment above is outdated after changes to improve partial lengths asymptotics.

Reply:
I'm also not sure how any changes to partial lengths would affect this, but I would also have thought that the logic for the case where segment and offset are undefined is needed. I think in the case where a slide is required, getContainingSegment would give an undefined segment back and hit the assert before getting to the slide call.

Reply:
The fuzz tests are confirmed passing. It seems safe to remove the undefined validation

Reply:
ok. I believe it's been made outdated; that code predates Tony's refactor of merge-tree walks as well as mine of leveraging partial lengths in reconnect.